packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol

```diff
@@ -61,27 +335,325 @@ library ConnextUtils {
+    // Sanity check: to is sensible
+    if (_recipient == address(0)) revert ConnextUtils__removeLiquidity_recipientEmpty();
+
+    // Sanity check: nonzero amounts
+    if (_amount == 0) revert ConnextUtils__removeLiquidity_amountIsZero();
+
+    uint256 routerBalance = _routerBalances[msg.sender][_local];
+    // Sanity check: amount can be deducted for the router
+    if (routerBalance < _amount) revert ConnextUtils__removeLiquidity_insufficientFunds();
+
+    // Update router balances
+    unchecked {
+      _routerBalances[msg.sender][_local] = routerBalance - _amount;
+    }
+
+    // Transfer from contract to specified to
+    _transferAssetFromContract(_local, _recipient, _amount, _wrapper);
+
+    // Emit event
+    emit LiquidityRemoved(msg.sender, _recipient, _local, _amount, msg.sender);
+  }
+
+  function xcall(
+    xCallLibArgs calldata _args,
+    mapping(address => ConnextMessage.TokenId) storage _adoptedToCanonical,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools,
+    mapping(bytes32 => uint256) storage _relayerFees
+  ) external returns (bytes32, uint256) {
+    _xcallSanityChecks(_args);
+
+    // get the true transacting asset id (using wrapped native instead native)
+    (bytes32 transferId, bytes memory message, xCalledEventArgs memory eventArgs) = _processXcall(
+      _args,
+      _adoptedToCanonical,
+      _adoptedToLocalPools
+    );
+
+    // Store the relayer fee
+    _relayerFees[transferId] = _args.xCallArgs.relayerFee;
+
+    // emit event
+    emit XCalled(transferId, _args.xCallArgs, eventArgs, _args.nonce, message, msg.sender);
+
+    return (transferId, _args.nonce + 1);
+  }
+
+  /**
+   * @notice Called on the destination domain to disburse correct assets to end recipient
+   * and execute any included calldata
+   * @dev Can be called prior to or after `handle`, depending if fast liquidity is being
+   * used.
+   */
+  function execute(
+    ExecuteLibArgs calldata _args,
+    mapping(bytes32 => address[]) storage _routedTransfers,
+    mapping(bytes32 => bool) storage _reconciledTransfers,
+    mapping(address => mapping(address => uint256)) storage _routerBalances,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools,
+    mapping(bytes32 => address) storage canonicalToAdopted,
+    RouterPermissionsManagerInfo storage _routerInfo,
+    mapping(bytes32 => address) storage _transferRelayer
+  ) external returns (bytes32 transferId) {
+    (bytes32 transferId, bool reconciled) = _executeSanityChecks(_args, _transferRelayer, _reconciledTransfers);
+
+    // check to see if the transfer was reconciled
+    (uint256 amount, address adopted) = _handleExecuteLiquidity(
+      transferId,
+      !reconciled,
+      _args,
+      _routedTransfers,
+      _routerBalances,
+      _adoptedToLocalPools,
+      canonicalToAdopted,
+      _routerInfo
+    );
+
+    // execute the transaction
+    _handleExecuteTransaction(_args, amount, adopted, transferId, reconciled);
```

Comment:
in keeping up with the naming convention for extending functions (functions we're using to avoid stack too deep errors), maybe we could change `_handleExecuteTransaction` to `_executeProcess` ?

Reply:
theres some renaming that will happen in a separate PR. this will really only be relevant for `xcall`, `execute`, and `reconcile` -- basically im thinking:

- the `ConnextHandler` has the top level functions
- the `ConnextLogic` has the same top level function names
- if there are smaller functions within the `ConnextLogic` function, it should be named `_process${TOP_LEVEL_NAME}${DESCRIPTOR_OF_PROCESS}`. could also be named `_handle{...}` with the same structure

we can do this in other PRs